tests/source/issue-3851-4502.rs

```diff
@@ -19,4 +19,9 @@ fn main() {
     if true {
```

Comment:
let's leave this file name and contents as-is, and create a new source/target file pair for issue 4502 to be consistent with the rest of the codebase. The two patterns we like to stick with are separate, specific test files for individual issues that make it easy to search/locate tests for a specific issue with a clear picture on that specific issue, or broader/catch-all types of files that may have snippets/cases from various issues that are delineated with things like inline comments or function names

Reply:
Submitted.  Split the tests as requested.